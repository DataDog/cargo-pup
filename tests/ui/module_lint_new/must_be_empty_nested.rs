//@compile-flags: --crate-name test_must_be_empty_nested
//@compile-flags: --crate-type lib

// Parent module (not targeted by lint, allowed to have content)
pub mod parent_module {
    // This module can have content, it's not targeted
    pub fn allowed_function() {
        println!("This is allowed in the parent");
    }
    
    // Nested module that IS targeted by the lint (should trigger an error)
    pub mod nested_target { //~ ERROR: Module must be empty
        pub fn invalid_function() {
            println!("This shouldn't be here");
        }
        
        pub struct InvalidStruct;
    }
    
    // Another nested module (not targeted, allowed to have content)
    pub mod another_nested {
        pub const ALLOWED_CONST: &str = "This is fine";
    }
} 