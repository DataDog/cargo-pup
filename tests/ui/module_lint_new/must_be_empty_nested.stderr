error: Item 'invalid_function' not allowed in empty module
  --> tests/ui/module_lint_new/must_be_empty_nested.rs:15:9
   |
LL | /         pub fn invalid_function() {
LL | |             println!("This shouldn't be here");
LL | |         }
   | |_________^
   |
   = help: Remove this item from the module, which must be empty
   = note: Applied by cargo-pup rule 'must_be_empty_nested_test'.
   = note: `#[deny(module_must_be_empty)]` on by default

error: Item 'InvalidStruct' not allowed in empty module
  --> tests/ui/module_lint_new/must_be_empty_nested.rs:19:9
   |
LL |         pub struct InvalidStruct;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Remove this item from the module, which must be empty
   = note: Applied by cargo-pup rule 'must_be_empty_nested_test'.

error: aborting due to 2 previous errors

